infra.bs

```diff
@@ -608,58 +612,60 @@ U+007A (z), inclusive.
 
-<p>A <a>JavaScript string</a>'s <dfn export for="JavaScript string">length</dfn> is the number of
-<a>code units</a> it contains.
-
-<p>A <a>JavaScript string</a> can also be interpreted as containing <a>code points</a>, per the
-conversion defined in <a>The String Type</a> section of the JavaScript specification. [[!ECMA-262]]
+<p>A <a>string</a> can also be interpreted as containing <a>code points</a>, per the conversion
+defined in <a>The String Type</a> section of the JavaScript specification. [[!ECMA-262]]
 
 <p class=note>This conversion process converts surrogate pairs into their corresponding
 <a>scalar value</a> and maps isolated surrogates to their corresponding <a>code point</a>, leaving
 them effectively as-is.
 
-<p class=example id=example-javascript-string-in-code-points>A <a>JavaScript string</a> consisting
-of the <a>code units</a> 0xD83D, 0xDCA9, and 0xD800, when interpreted as containing
-<a>code points</a>, would consist of the <a>code points</a> U+1F4A9 and U+D800.
+<p class=example id=example-javascript-string-in-code-points>A <a>string</a> consisting of the
+<a>code units</a> 0xD83D, 0xDCA9, and 0xD800, when interpreted as containing <a>code points</a>,
+would consist of the <a>code points</a> U+1F4A9 and U+D800.
+
+<p>A <a>string</a>'s
+<dfn export for="string,JavaScript string,scalar value string" id=javascript-string-length>length</dfn>
+is the number of <a>code units</a> it contains.
 
-<p>A <dfn export>scalar value string</dfn> is a sequence of <a>scalar values</a>.
+<p>A <a>string</a>'s
+<dfn export for="string,JavaScript string,scalar value string">code point length</dfn> is the number
+of <a>code points</a> it contains.
+
+<p>A <dfn export>scalar value string</dfn> is a <a>string</a> whose <a>code points</a> are all
+<a>scalar values</a>.
 
 <p class=note>A <a>scalar value string</a> is useful for any kind of I/O or other kind of operation
 where <a>UTF-8 encode</a> comes into play.
 <!-- It's also useful if you can imagine the subsystem to be implemented in Rust -->
 
-<p><dfn export lt=string>String</dfn> can be used to refer to either a <a>JavaScript string</a> or
-<a>scalar value string</a>, when it is clear from the context which is meant or when the distinction
-is immaterial. <a>Strings</a> are denoted by double quotes and monospace font.
-
-<p class=example id=example-string-notation>"<code>Hello, world!</code>" is a string.
-
-<p>A <a>string</a>'s <dfn export for=string>length</dfn> is the number of <a>code points</a> it
-contains.
-
-<p>To <dfn export for="JavaScript string">convert</dfn> a <a>JavaScript string</a> into a
-<a>scalar value string</a>, replace any <a>surrogates</a> with U+FFFD.
-<!-- Obviates need for https://heycam.github.io/webidl/#dfn-obtain-unicode -->
+<p>To <dfn export for="string,JavaScript string" id=javascript-string-convert>convert</dfn> a
+<a>string</a> into a <a>scalar value string</a>, replace any <a>surrogates</a> with U+FFFD.
```

Comment:
Should this say "unpaired surrogates" or "surrogate code points", since a string/JavaScript string can contain surrogate code *units*? I see that the following note covers this, but maybe calling out code points here would help?

Reply:
I'm going to leave this as-is since surrogate is defined as a code point and U+FFFD is also a code point (the equivalent code unit would be denoted as 0xFFFD). If I keep getting questions about this though I'll reconsider.